File: CCCInventory/Customer.cs
using System.ComponentModel.DataAnnotations;

namespace CCCInventory
{
    public class Customer
    {
        [Key]
        public int CustomerId { get; set; }
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public List<Order> Orders { get; set; } = null!;
    }
}

File: CCCInventory/ClientApp/src/app/models/customer.ts
export class Customer {
  customerId?: number;
  firstName: string | undefined;
  lastName: string | undefined;
  email?: string;
  phone?: string;
  orders?: any[];
  orderCount?: number;
}

export class CustomerSearchResult {
  customerId: number | undefined;
  firstName: string | undefined;
  lastName: string | undefined;
  email?: string;
  phone?: string;
}
